libs/coin-modules/coin-bitcoin/src/wallet-btc/wallet.test.ts

```diff
@@ -0,0 +1,261 @@
+import BitcoinLikeWallet from "./wallet";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { Account } from "./account";
+import Xpub from "./xpub";
+import BitcoinLikeStorage from "./storage";
+import BitcoinLikeExplorer from "./explorer";
+import { PickingStrategy } from "./pickingstrategies/types";
+import BigNumber from "bignumber.js";
+import { TX, Address, Output } from "./storage/types";
+import { DerivationModes, TransactionInfo } from "./types";
+import { ICrypto } from "./crypto/types";
+
+jest.mock("./explorer");
+jest.mock("./crypto/factory");
+
+const DERIVATION_MODE = DerivationModes.TAPROOT;
+
+const mockStorage = new BitcoinLikeStorage();
+
+const mockCrypto = {
+  getAddress: jest.fn(),
+  toOutputScript: jest.fn(),
+  toOpReturnOutputScript: jest.fn(),
+  network: {
+    name: "testnet",
+  },
+} as unknown as jest.Mocked<ICrypto>;
+
+describe("BitcoinLikeWallet", () => {
+  let wallet: BitcoinLikeWallet;
+  let mockCryptoCurrency: CryptoCurrency;
+  let mockAccount: Account;
+
+  beforeEach(() => {
+    wallet = new BitcoinLikeWallet();
+
+    mockCryptoCurrency = {
+      type: "CryptoCurrency",
+      id: "bitcoin",
+      name: "Bitcoin",
+      family: "bitcoin",
+      coinType: 0,
+      scheme: "native_segwit",
+      units: [
+        {
+          name: "BTC",
+          code: "BTC",
+          magnitude: 8,
+        },
+      ],
+      explorerViews: [
+        {
+          tx: "https://blockchain.info/tx/{txid}",
+          address: "https://blockchain.info/address/{address}",
+        },
+      ],
+      ticker: "BTC",
+      explorerId: "btc",
+      color: "#FF0000",
+      managerAppName: "Bitcoin",
+    } as CryptoCurrency;
+
+    mockAccount = {
+      params: {
+        xpub: "test-xpub",
+        path: "test-path",
+        index: 0,
+        currency: "bitcoin",
+        network: "mainnet",
+        derivationMode: DERIVATION_MODE,
+      },
+
+      xpub: new Xpub({
+        storage: mockStorage,
+        explorer: new BitcoinLikeExplorer({ cryptoCurrency: mockCryptoCurrency }),
+        crypto: mockCrypto,
+        xpub: "test-xpub",
+        derivationMode: DERIVATION_MODE,
+      }),
+    } as Account;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
```

Comment:
Perhaps we need to add unit tests for the "signAccountTx" method

Reply:
Added a simple test on this one, might need to iterate over it but Im not sure it's worth it.